packages/tailwindcss/src/index.ts

```diff
@@ -381,15 +380,18 @@ async function parseCss(
 
       for (let keyframesRule of keyframesRules) {
         // Remove any keyframes that aren't used by an animation variable.
-        let keyframesName = keyframesRule.selector.slice(11) // `@keyframes `.length
+        let keyframesName = keyframesRule.params
         if (!animationParts.includes(keyframesName)) {
           continue
         }
 
         // Wrap `@keyframes` in `AtRoot` so they are hoisted out of `:root` when
         // printing.
         nodes.push(
-          Object.assign(keyframesRule, atRoot([rule(keyframesRule.selector, keyframesRule.nodes)])),
+          Object.assign(
+            keyframesRule,
+            atRoot([atRule(keyframesRule.name, keyframesRule.params, keyframesRule.nodes)]),
+          ),
         )
```

Comment:
Do you know if this can be `nodes.push(atRoot([keyframesRule]))`?

Reply:
Just ported it as-is, but if we just use `Object.assign(keyframesRule, atRoot([keyframesRule]))` then we get an infinite loop (maybe that's why we did it this way).

That said, the `nodes.push(…)` you mentioned seems to work as expected.